lib/travis/build/git/clone.rb

```diff
@@ -88,14 +88,19 @@ def delete_netrc
           end
 
           def github?
+            source_host_name.downcase == 'github.com' || source_host_name.downcase.end_with?('.github.com')
```

Comment:
Neither seems to indicate the intent clear. `source_host_name.downcase.end_with?('github.com')` is incorrect, as `evilgithub.com` would match. I want the last 2 levels of `source_host_name` to match `github` and `com`, _*exactly*_.

Maybe `source_host_name.split('.')[-2,2] == ['github', 'com']`, but that seems rather unsightly.